app/models/assignment.rb

```diff
@@ -13,23 +13,40 @@ class Assignment < ActiveRecord::Base
   validates :mixpanel_result, inclusion: { in: %w(success failure) }, allow_nil: true
   validate :variant_must_exist
 
-  scope :unsynced_to_mixpanel, -> { where("mixpanel_result = 'failure' OR mixpanel_result IS NULL") }
-
   normalize_attributes :mixpanel_result
 
   delegate :name, to: :split, prefix: true
 
-  class << self
-    def to_hash
-      Hash[all.includes(:split).map { |a| [a.split.name.to_sym, a.variant.to_sym] }]
-    end
+  scope :unsynced_to_mixpanel, -> { where("mixpanel_result = 'failure' OR mixpanel_result IS NULL") }
 
-    def for_presentation(built_at: nil)
-      q = joins(:split)
-        .where("splits.decided_at is null or assignments.updated_at > splits.decided_at")
-      q = built_at.nil? ? q.where("splits.finished_at is null") : q.where("splits.finished_at > ?", built_at)
-      q
-    end
+  scope :for_presentation, ->(app_build: nil) do
```

Comment:
Rather than pass a bag of disjunct args at this scope, I realized a type representing an app build was better - you either aren't presenting for a specific app build (and you get the default behavior for Rails/JS web clients), or you are, and you care about version, build date, and which app is in play. (theoretically you could have JS clients in the field at some point, which this version-aware code will enable too).

Reply:
BTW, my current hunch is that I'm going to eliminate the visitor-to-assignments association and replace it with a class method like so:

```ruby
def self.assignments(params = {})
  Assignment.where(split: self).for_presentation(params)
end
```

I just don't need a fully-loaded association. I might need association-cache aware memoization though, so that might get sticky.

Anyway, that's still for later.